test/lib/study_cleanup_tools_test.rb

```diff
@@ -0,0 +1,105 @@
+require 'test_helper'
+
+# test all validations to ensure security is being enforced that will block accidental deletes of user studies/workspaces
+# does not directly test cleanup methods as these would destroy seed data needed for other tests
+class StudyCleanupToolsTest < ActiveSupport::TestCase
+
+  include Minitest::Hooks
+  include SelfCleaningSuite
+  include TestInstrumentor
+
+  before(:all) do
+    @user = FactoryBot.create(:user, test_array: @@users_to_clean)
+    @basic_study = FactoryBot.create(:study, name_prefix: 'Cleanup Security Checks', test_array: @@studies_to_clean)
+  end
+
+  teardown do
+    Rails.env = "test" # reset this value to prevent other env checks failing later as this will persist across all runs
+  end
+
+  test 'should validate hostname' do
+    assert StudyCleanupTools.permit_hostname?
+  end
+
+  test 'should validate billing project' do
+    billing_project = FireCloudClient::PORTAL_NAMESPACE
+    assert StudyCleanupTools.permit_billing_project?(billing_project)
+
+    bad_project = "this-is-not-valid"
+    refute StudyCleanupTools.permit_billing_project?(bad_project)
+  end
+
+  test 'should validate environment' do
+    assert StudyCleanupTools.permit_environment?
+
+    # test allowing development environment
+    Rails.env = 'development'
+    assert StudyCleanupTools.permit_environment?(allow_dev_env: true)
+
+    # ensure production-like environments are disallowed
+    Rails.env = 'staging'
+    refute StudyCleanupTools.permit_environment?
```

Comment:
does this test need to reset the env at the end?  I'd also rather stubbing was used so that the environment isn't left in a weird state if the test fails midway through

Reply:
the `teardown` declaration resets it.  Since `permit_environment?` calls `Rails.env` directly, I'm not sure this can be stubbed, but I'll take a crack at it.

Reply:
32c1bc3